flint.ui/src/store/index.js

```diff
@@ -1,220 +1,409 @@
+            },
+          },
+        },
+        {
+          initSoil: {
+            dpmaCMInit: "#$0.02#$",
+            rpmaCMInit: "#$8.681#$",
+            biofCMInit: "#$0.05#$",
+            biosCMInit: "#$0.03#$",
+            humsCMInit: "#$20.96#$",
+            inrtCMInit: "#$43.0#$",
+            TSMDInit: "#$35.21#$",
+          },
+        },
+        {
+          soil: {
+            fracManuCMToDpma: "#$0.49#$",
+            fracManuCMToRpma: "#$0.49#$",
+            fracManuCMToBiof: "#$0.0#$",
+            fracManuCMToBios: "#$0.0#$",
+            dToRRatioInPres: "#$2.84#$",
+            encpFracHums: "#$0.68#$",
+            sampleDepth: "#$26.0#$",
+            sdcmRateMultDpm: "#$12.0#$",
+            sdcmRateMultRpm: "#$0.5#$",
+            sdcmRateMultBiof: "#$0.0#$",
+            sdcmRateMultBios: "#$0.76#$",
+            sdcmRateMultHums: "#$0.05#$",
+            fracPbioToBiof: "#$0.49#$",
+            fracHumsToBios: "#$0.0#$",
+            evapoOpenRatio: "#$2.5#$",
+            bToCMaxTSMDRatio: "#$3.0#$",
+          },
+        },
+      ],
+      Modules: {
+        TransactionManagerEndOfStepModule: {
+          library: "internal.flint",
+          order: 1,
+        },
+        CalendarSequencer: {
+          library: "internal.flint",
+          order: 2,
+        },
+        PlantResidueModule: {
+          enabled: true,
+          library: "moja.flint.example.rothc",
+          order: 3,
+        },
+        SoilCoverModule: {
+          enabled: true,
+          library: "moja.flint.example.rothc",
+          order: 4,
+        },
+        RothCModule: {
+          enabled: true,
+          library: "moja.flint.example.rothc",
+          order: 5,
+        },
+        ErrorScreenWriter: {
+          enabled: true,
+          library: "moja.flint.example.base",
+          order: 10,
+          settings: {
+            log_errors_max_number: 10,
+          },
+        },
+        OutputerStream: {
+          enabled: true,
+          library: "internal.flint",
+          order: 11,
+          settings: {
+            output_filename: "Example_Rothc_Point_Stock.csv",
+            output_to_screen: true,
+            output_info_header: true,
+          },
+        },
+        OutputerStreamFlux: {
+          enabled: true,
+          library: "internal.flint",
+          order: 12,
+          settings: {
+            output_filename: "Example_Rothc_Point_Flux.csv",
+            output_to_screen: false,
+            output_info_header: true,
+          },
+        },
+      },
     },
-    actions: {
-
+  },
+  mutations: {
+    // soil_func_data_orig (state) {
+    // state.Variables[8].rainfall.transform.data_orig = "cool";
+    // console.log(state);
+    // },
+    setNewConfig(state, newValue) {
+      state.Variables[8].rainfall.transform.data_orig = newValue;
     },
-    modules: {
-
-    }
-
-})
\ No newline at end of file
+    set_pointConfig_Pool_1(state, pool_1_value) {
```

Comment:
Can you generalise this by also passing in the Pool index?

Reply:
I did try that, but a single mutation makes only one change at a time. If we try to do multiple changes in the state, then we get the changes being passed as _'undefined'_.

![image](https://user-images.githubusercontent.com/13800316/127241724-bb30231e-b9ad-4f96-9a12-76bba108e9d4.png)